internal/ent/hooks/trustcenterdoc.go

```diff
@@ -151,10 +156,22 @@ func HookUpdateTrustCenterDoc() ent.Hook { // nolint:gocyclo
 
 			logx.FromContext(ctx).Debug().Msg("trust center doc hook")
 
-			// if the watermark status is true already, we do not want to be able to revert it under any circumstances
-			if _, ok := m.WatermarkingEnabled(); ok {
-				if oldWatermarkingEnabled, err := m.OldWatermarkingEnabled(ctx); err == nil && oldWatermarkingEnabled {
-					m.ResetWatermarkingEnabled()
+			// Check if watermarking is being modified and enforce global config override rules
+			if newWatermarkingEnabled, watermarkingEnabledSet := m.WatermarkingEnabled(); watermarkingEnabledSet {
+				// Get the trust center ID - for updates we need to look it up from the document
+				docID, docIDSet := m.ID()
+				if docIDSet {
+					doc, err := m.Client().TrustCenterDoc.Get(ctx, docID)
+					if err == nil && doc != nil {
+						config, err := m.Client().TrustCenterWatermarkConfig.Query().Where(trustcenterwatermarkconfig.TrustCenterID(doc.TrustCenterID)).Only(ctx)
+						if err == nil && config != nil {
+							// If global config is enabled, force document-level watermarking to true
+							if config.IsEnabled && !newWatermarkingEnabled {
+								logx.FromContext(ctx).Debug().Msg("global watermark config is enabled, forcing document watermarking to true")
+								m.SetWatermarkingEnabled(true)
+							}
+						}
+					}
```

Comment:
It feels like this is being written for a case when an organization would ever have more than one trust center or watermark config which isn't true. I think we should fix the tests to reflect reality which would simplify this logic drastically to 

```suggestion
						config, err := m.Client().TrustCenterWatermarkConfig.Query().Only(ctx)
						if err == nil && config != nil {
							// If global config is enabled, force document-level watermarking to true
							if config.IsEnabled && !newWatermarkingEnabled {
								logx.FromContext(ctx).Debug().Msg("global watermark config is enabled, forcing document watermarking to true")
								m.SetWatermarkingEnabled(true)
							}
						}
					
```